src/write_graphite.c

```diff
@@ -297,10 +297,9 @@ static void wg_callback_free (void *data)
 
     pthread_mutex_lock (&cb->send_lock);
 
-    wg_flush_nolock (/* timeout = */ 0, cb);
-
     if (cb->sock_fd >= 0)
     {
+        wg_flush_nolock (/* timeout = */ 0, cb);
```

Comment:
I think a check in `wg_send_buffer` makes even more sense. That is, check cb->sock_fd where it's actually used instead of letting callers worry about it.